blog/_posts/2023-05-26-scala-3.3.0-released.md

```diff
@@ -0,0 +1,162 @@
+
+- `-Wunused:imports` - for unused imports
+- `-Wunused:privates` - for unused local definitions
+- `-Wunused:locals`  - for unused local definitions
+- `-Wunused:explicits` - for unused explicit parameters
+- `-Wunused:implicits` - for unused implicit parameters (parameters in `using` clauses)
+- `-Wunused:params` - for all unused method parameters
+- `-Wunused:all` - for enabling all of the above lints
+
+#### Checking for discarded values
+
+Discarding non-unit values is usually the symptom of subtle mistakes. The compiler can now warn you about all discarded values saving you from bugs resulting from those hard-to-spot mistakes.
+
+In the following simplified example
+
+```Scala
+def failure: Either[String, Nothing] = Left("something broke")
+
+def failedComputation: Either[String, Unit] =
+  Right(()).map(_ => failure)
+```
+
+the programmer, by mistake, used the `map` method instead of `flatMap`. The code still compiles, but due to value discarding, behaves unexpectedly, returning `Right(())` from the `failedComputation` method. If the `-Wvalue-discard` flag is enabled, the compiler will report the warning, saving the user from a potential bug.
+
+### More consistent brace-less syntax
+
+Braces around method parameters can now be replaced with a colon. This can lead to cleaner, shorter, and often more readable code in places like configuration DSLs or test case definitions.
+
+```Scala
+def canFail(input: String) = Try:
+  someComputation(input).flatMap: res =>
+    val partial = moreComputation(res)
+    andEvenMore(partial)
+
+class TestSuite extends munit.FunSuite:
+  test("the job doesn't fail"):
+    val job = canFail("some data")
+    assert(job.isSuccess)
+```
+
+### `boundary` and `break`
+
+Two new methods were added to the standard library in the place: `boundary` and `brake`. They are safer and more expressive replacements for recently deprecated non-local returns.
+
+`break` allows for a type-safe early escape from anywhere inside the block delimited by `boundary` to its end, returning the passed value from the entire block.
+
+```Scala
+import util.boundary, boundary.break
+
+def sumOfRoots(number: List[Double]): Option[Double] = boundary {
+  val roots = numbers.map { n =>
+    println(s" * calculating square root for $n*")
+    if n >= 0 then Math.sqrt(n) else break(None)
+  }
+  Some(roots.sum)
+}
+```
+
+When you run the above method, you will notice that it returns None when any of the input elements is negative. Moreover, thanks to `boundary`/`break`, you can see from the console output that it stops the entire process after encountering the first negative element.
+
+`break` can jump out of the`boundary` in the function deeper on the stack. To make it safe, only functions with a matching `Label` in their using clauses can break. This can be used to create isolated parts of an application with streamlined error handling.
```

Comment:
```suggestion
`break` can jump out of the `boundary` in the function deeper on the stack. To make it safe, only functions with a matching `Label` in their using clauses can break. This can be used to create isolated parts of an application with streamlined error handling.
```